File: entity/message.go
package entity

import "time"

type Message struct {
	ID        int `gorm:"primaryKey"`
	Username  string
	User      User `gorm:"foreignKey:Username"`
	GroupID   int
	Group     Group  `gorm:"foreignKey:GroupID"`
	Content   string `gorm:"size:255"`
	CreatedAt time.Time
}

File: frontend/src/types/message.ts
export interface GroupChat {
  id: number;
  name: string;
  isDirect: boolean;
  ownerName: string;
  members: string[];
  lastMessage: Message | null;

  avatar?: string;
}

export interface Message {
  id: number;
  username: string;
  groupId: number;
  content: string;
  createdAt: string;
}